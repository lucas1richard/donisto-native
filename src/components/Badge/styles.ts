import { StyleSheet } from 'react-native';

const badgeStyles = StyleSheet.create({
  wrapper: {
    borderRadius: 999,
    paddingHorizontal: 10,
    paddingVertical: 3,
    backgroundColor: 'rgba(0,0,0,0.6)',
  },
});

export default badgeStyles;