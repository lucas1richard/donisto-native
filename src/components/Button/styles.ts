import { StyleSheet, TextStyle, ViewStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultViewStyles: ViewStyle = {
  // padding: theme.fontSize,
  height: 46,
  width: '100%',
  borderRadius: 8,
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'row'
};

export const viewStyles = StyleSheet.create({
  outlined: {
    ...defaultViewStyles,
    borderWidth: 1,
    borderColor: theme.primary[600],
    backgroundColor: theme.primary[50]
  },
  rounded: {
    ...defaultViewStyles,
    borderRadius: 99
  },
  contained: {
    ...defaultViewStyles
  },
  text: {
  },
  primary: {
    ...defaultViewStyles,
    backgroundColor: theme.primary[500]
  },
  secondary: {
    ...defaultViewStyles,
    backgroundColor: theme.secondary[400]
  },
  default: {
    ...defaultViewStyles,
    backgroundColor: '#f2f2f2'
  },
  _default: {
    ...defaultViewStyles,
    backgroundColor: '#f2f2f2'
  }
});

const defaultTextStyles: TextStyle = {
  fontSize: theme.fs
};

export const textStyles = StyleSheet.create({
  'text-primary': {
    color: theme.primary[400],
    borderBottomWidth: 1,
    borderBottomColor: theme.primary[400]
  },
  'text-secondary': {
    color: theme.secondary[400]
  },
  'text-default': {
    color: '#000'
  },
  primary: {
    ...defaultTextStyles,
    color: '#fff'
  },
  secondary: {
    ...defaultTextStyles,
    color: '#fff'
  },
  default: {
    ...defaultTextStyles,
    color: '#000'
  },
  _default: {
    ...defaultTextStyles,
    color: '#000'
  }
});
