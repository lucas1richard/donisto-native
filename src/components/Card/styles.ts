import theme from 'theme/iftheme';
import { StyleSheet } from 'react-native';

/**
 * Function which creates an object to be inserted into StyleSheet
 */

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: '#fff',
    marginTop: theme.fs,
    padding: theme.fs,
    elevation: 3,
    shadowColor: '#555',
    shadowOpacity: 0.15,
    shadowOffset: {
      width: 2,
      height: 4
    },
    shadowRadius: 6
  }
});

export default styles;
