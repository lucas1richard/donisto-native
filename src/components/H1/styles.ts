import { StyleSheet, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultStyles: TextStyle = {
  fontSize: theme.fontSize * 2,
  color: theme.black,
  textAlign: 'center'
};

export default StyleSheet.create({
  default: {
    ...defaultStyles
  },
  primary: {
    ...defaultStyles,
    color: theme.primary[400]
  },
  secondary: {
    ...defaultStyles,
    color: theme.secondary[400]
  },
  light: {
    ...defaultStyles,
    color: '#fff'
  }
});
