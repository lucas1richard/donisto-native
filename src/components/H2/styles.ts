import { StyleSheet, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultStyles: TextStyle = {
  fontSize: theme.fontSize * 1.7,
  color: theme.black,
  textAlign: 'center'
};

export default StyleSheet.create({
  default: {
    ...defaultStyles
  },
  primary: {
    ...defaultStyles,
    color: theme.primary[500]
  },
  secondary: {
    ...defaultStyles,
    color: theme.secondary[500]
  },
  light: {
    ...defaultStyles,
    color: '#fff'
  }
});
