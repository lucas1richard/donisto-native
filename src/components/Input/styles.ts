import { StyleSheet, ViewStyle, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

export const style = {
  wrapper: {
    backgroundColor: '#fff',
    width: '100%'
  },
  errortrue: error(true),
  errorfalse: error(false),
  fieldLabel: fieldLabel(),
  fieldLabelerror: fieldLabel({ color: theme.red }),
  fieldLabelvalid: fieldLabel({ color: theme.green }),
  fieldLabelactive: fieldLabel({ color: theme.darkBlue }),
  textInputWrapper: textInputWrapper({ borderColor: theme.pristineInputGray }),
  textInputWrappererror: textInputWrapper({ borderColor: theme.red }),
  textInputWrappervalid: textInputWrapper({ borderColor: theme.green }),
  textInputWrapperactive: textInputWrapper({ borderColor: theme.darkBlue }),
  inputStyle: inputStyle(),
  inputStyleerror: inputStyle({ color: theme.red }),
  inputStylevalid: inputStyle({ color: theme.green }),
  inputStyleactive: inputStyle({ color: theme.darkBlue })
};

export default StyleSheet.create(style);

function inputStyle(addStyle = {}): TextStyle {
  return {
    width: '100%',
    color: theme.darkGray,
    position: 'relative',
    padding: 12,
    fontFamily: 'proxima-soft-regular',
    fontWeight: '300',
    height: 120,
    fontSize: 18,
    marginTop: -35,
    ...addStyle
  };
}

function textInputWrapper(addStyle = {}): ViewStyle {
  return {
    borderRadius: 4,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: theme.darkGray,
    backgroundColor: '#fff',
    width: '100%',
    height: 50,
    overflow: 'hidden',
    ...addStyle
  };
}

/** The input label */
function fieldLabel(addStyle = {}): TextStyle {
  return ({
    color: theme.darkGray,
    backgroundColor: 'rgba(0,0,0,0)',
    marginBottom: 3,
    fontWeight: '300',
    fontSize: 12,
    ...addStyle
  });
}

/**
 * Error text style
 */
function error(err: boolean): TextStyle {
  return {
    color: theme.red,
    fontSize: 12,
    height: 16,
    opacity: err ? 1 : 0
  };
}
