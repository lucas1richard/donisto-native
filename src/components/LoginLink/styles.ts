import { StyleSheet } from 'react-native';
import theme from 'theme/iftheme';

export default StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: theme.fs * 2
  },
  loginWrapper: {
    borderBottomWidth: theme.fs * 0.05,
    borderBottomColor: '#fff'
  }
});