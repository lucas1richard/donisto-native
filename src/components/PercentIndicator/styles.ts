import { StyleSheet } from 'react-native';
import theme from 'theme/iftheme';

const pIndicatorStyles = StyleSheet.create({
  wrapper: {
    height: theme.fontSize / 2,
    backgroundColor: '#dedede',
    overflow: 'hidden',
    width: '100%'
  },
  bar: {
    backgroundColor: theme.primary[500],
    height: '100%'
  }
});

export default pIndicatorStyles;