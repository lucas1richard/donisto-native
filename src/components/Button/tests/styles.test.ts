import { textStyles, viewStyles } from '../styles';

describe('button styles', () => {
  it('has text-styles', () => {
    expect(textStyles).toMatchSnapshot();
  });
  it('has view-styles', () => {
    expect(viewStyles).toMatchSnapshot();
  });
});