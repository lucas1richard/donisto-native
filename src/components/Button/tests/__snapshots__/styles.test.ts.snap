// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button styles has text-styles 1`] = `
Object {
  "_default": Object {
    "color": "#000",
    "fontSize": 20,
  },
  "default": Object {
    "color": "#000",
    "fontSize": 20,
  },
  "primary": Object {
    "color": "#fff",
    "fontSize": 20,
  },
  "secondary": Object {
    "color": "#fff",
    "fontSize": 20,
  },
  "text-default": Object {
    "color": "#000",
  },
  "text-primary": Object {
    "borderBottomColor": "rgb(238, 135, 119)",
    "borderBottomWidth": 1,
    "color": "rgb(238, 135, 119)",
  },
  "text-secondary": Object {
    "color": "#ffa726",
  },
}
`;

exports[`button styles has view-styles 1`] = `
Object {
  "_default": Object {
    "alignItems": "center",
    "backgroundColor": "#f2f2f2",
    "borderRadius": 8,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "contained": Object {
    "alignItems": "center",
    "borderRadius": 8,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "default": Object {
    "alignItems": "center",
    "backgroundColor": "#f2f2f2",
    "borderRadius": 8,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "outlined": Object {
    "alignItems": "center",
    "backgroundColor": "#fcebe8",
    "borderColor": "#3949ab",
    "borderRadius": 8,
    "borderWidth": 1,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "primary": Object {
    "alignItems": "center",
    "backgroundColor": "rgb(251, 100, 65)",
    "borderRadius": 8,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "rounded": Object {
    "alignItems": "center",
    "borderRadius": 99,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "secondary": Object {
    "alignItems": "center",
    "backgroundColor": "#ffa726",
    "borderRadius": 8,
    "flexDirection": "row",
    "height": 46,
    "justifyContent": "center",
    "width": "100%",
  },
  "text": Object {},
}
`;
